pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -115,23 +124,35 @@ func desiredLoadBalancerService(ci *operatorv1.IngressController, deploymentRef
 	if service.Annotations == nil {
 		service.Annotations = map[string]string{}
 	}
-	if infraConfig.Status.Platform == configv1.AWSPlatformType {
-		service.Annotations[awsLBProxyProtocolAnnotation] = "*"
-		// Set the load balancer for AWS to be as aggressive as Azure (2 fail @ 5s interval, 2 healthy)
-		service.Annotations["service.beta.kubernetes.io/aws-load-balancer-healthcheck-interval"] = "5"
-		service.Annotations["service.beta.kubernetes.io/aws-load-balancer-healthcheck-timeout"] = "4"
-		service.Annotations["service.beta.kubernetes.io/aws-load-balancer-healthcheck-unhealthy-threshold"] = "2"
-		service.Annotations["service.beta.kubernetes.io/aws-load-balancer-healthcheck-healthy-threshold"] = "2"
-	}
-	// Azure load balancers are not customizable and are set to (2 fail @ 5s interval, 2 healthy)
-	// GCP load balancers are not customizable and are set to (3 fail @ 8s interval, 1 healthy)
-
-	if infraConfig.Status.Platform == configv1.IBMCloudPlatformType {
-		service.Annotations[iksLBScopeAnnotation] = "public"
+	if infraConfig.Status.PlatformStatus != nil {
+		switch infraConfig.Status.PlatformStatus.Type {
+		case configv1.AWSPlatformType:
+			if ci.Status.EndpointPublishingStrategy.LoadBalancer == nil ||
+				ci.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters == nil ||
+				ci.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.Type == "AWS" &&
+					ci.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.Type == operatorv1.AWSClassicLoadBalancer {
+				service.Annotations[awsLBProxyProtocolAnnotation] = "*"
+			}
+			if ci.Status.EndpointPublishingStrategy.LoadBalancer != nil &&
+				ci.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil &&
+				ci.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.Type == "AWS" &&
+				ci.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.Type == operatorv1.AWSNetworkLoadBalancer {
+				service.Annotations[AWSLBTypeAnnotation] = AWSNLBAnnotation
+			}
+			// Set the load balancer for AWS to be as aggressive as Azure (2 fail @ 5s interval, 2 healthy)
+			service.Annotations["service.beta.kubernetes.io/aws-load-balancer-healthcheck-interval"] = "5"
+			service.Annotations["service.beta.kubernetes.io/aws-load-balancer-healthcheck-timeout"] = "4"
+			service.Annotations["service.beta.kubernetes.io/aws-load-balancer-healthcheck-unhealthy-threshold"] = "2"
+			service.Annotations["service.beta.kubernetes.io/aws-load-balancer-healthcheck-healthy-threshold"] = "2"
+		case configv1.IBMCloudPlatformType:
+			service.Annotations[iksLBScopeAnnotation] = "public"
+		}
+		// Azure load balancers are not customizable and are set to (2 fail @ 5s interval, 2 healthy)
```

Comment:
Could you provide a URL to documentation, etc, about these settings in the comment?

Reply:
@sgreene570 hese are godocs that [previously existed](https://github.com/openshift/cluster-ingress-operator/pull/427/files#diff-d75a10107f24ad23da316b5f7fad2c29L126-L127).

Reply:
Thanks, sounds good to me.